pkg/model/components/context.go

```diff
@@ -124,14 +124,24 @@ func IsBaseURL(kubernetesVersion string) bool {
 }
 
 // Image returns the docker image name for the specified component
-func Image(component string, clusterSpec *kops.ClusterSpec) (string, error) {
+func Image(component string, clusterSpec *kops.ClusterSpec, assetsBuilder *assets.AssetBuilder) (string, error) {
+	if assetsBuilder == nil {
+		return "", fmt.Errorf("unable to parse assets as assetBuilder is not defined")
+	}
+	// TODO remove this, as it is an addon now
 	if component == "kube-dns" {
 		// TODO: Once we are shipping different versions, start to use them
 		return "gcr.io/google_containers/kubedns-amd64:1.3", nil
 	}
 
 	if !IsBaseURL(clusterSpec.KubernetesVersion) {
-		return "gcr.io/google_containers/" + component + ":" + "v" + clusterSpec.KubernetesVersion, nil
+		image := "gcr.io/google_containers/" + component + ":" + "v" + clusterSpec.KubernetesVersion
+
+		image, err := assetsBuilder.RemapImage(image)
+		if err != nil {
+			return "", fmt.Errorf("unable to remap container %q: %v", image, err)
+		}
+		return image, nil
 	}
 
 	baseURL := clusterSpec.KubernetesVersion
```

Comment:
Maybe a comment here about why we're not remapping this case.  (It's not self-evident to me why we are not.)